### PR TITLE
Enable root7 and c++14 on all all platforms, excluding centos7

### PR DESCRIPTION
**Pull request by linev**

### CODE REVIEW

File: jenkins/root-build.cmake [hunk 1]
```diff
@@ -122,6 +119,16 @@ function(GET_ALL_SUPPORTED_MODULES_WIN32)
       unuran
     )
   endif()
+  if ("${ROOT_VERSION}" VERSION_GREATER "6.17")
+    list(APPEND all_supported
+      builtin_tbb
+      builtin_unuran
+      imt
+      unuran
```
Comment (Axel-Naumann) [inline jenkins/root-build.cmake]: Why the changes for builtin_tbb builtin_unuran imt unuran?

File: jenkins/root-build.cmake [hunk 2]
```diff
@@ -223,6 +230,12 @@ function(GET_ALL_SUPPORTED_MODULES_APPLE)
     builtin_xxhash
   )
 
+  if ("${ROOT_VERSION}" VERSION_GREATER "6.17")
+    list(APPEND all_supported
+      root7
+      webgui
+    )
+
```
Comment (Axel-Naumann) [inline jenkins/root-build.cmake]: Missing `endif()`

File: jenkins/root-build.cmake [hunk 3]
```diff
@@ -223,6 +230,12 @@ function(GET_ALL_SUPPORTED_MODULES_APPLE)
     builtin_xxhash
   )
 
+  if ("${ROOT_VERSION}" VERSION_GREATER "6.17")
```
Comment (Axel-Naumann) [inline jenkins/root-build.cmake]: Why do we need this - even without we only have `root7` and `webgui` enabled if the options exist.

File: jenkins/root-build.cmake [hunk 4]
```diff
@@ -326,6 +340,8 @@ function(GET_ALL_SUPPORTED_MODULES_LINUX)
   else()
     list(APPEND all_supported
       pythia8
+      root7
```
Comment (Axel-Naumann) [inline jenkins/root-build.cmake]: I believe that's the only change needed.

File: jenkins/root-build.cmake [hunk 5]
```diff
@@ -696,6 +712,8 @@ function(CONFIGURE_ROOT_OPTIONS)
       set(specflags ${specflags} -DCMAKE_CXX_STANDARD=17)
     elseif("cxx14" IN_LIST SPECLIST)
       set(specflags ${specflags} -DCMAKE_CXX_STANDARD=14)
+    elseif("root7" IN_LIST want_modules)
+      set(specflags ${specflags} -DCMAKE_CXX_STANDARD=14)
```
Comment (Axel-Naumann) [inline jenkins/root-build.cmake]: Does this mean that any build but CentOS7 will be built with C++14?

Comment (Axel-Naumann) [inline jenkins/root-build.cmake]: What happens if the build was configured to be C++17, is this going to override it to C++14 then?

**Comment by Axel-Naumann** [comment]
We need to keep C++ standard version the main selection knob, enabling root7 only if C++14 or later is on. With that, can you summarize the advantage of this PR?

File: jenkins/root-build.cmake (hunk 5 above)
Comment (linev) [reply jenkins/root-build.cmake]: Seems to be yes. 
I have to add extra check if C++ standard was already specified

Comment (linev) [reply jenkins/root-build.cmake]: Idea that root7 enabled on all platforms where it can be build. Centos will be excluded.
And only when root7 is where, one have to enable c++14. 
And seems to be I have to check, when c++ standard already specified

**Comment by linev** [comment]
> With that, can you summarize the advantage of this PR?

Idea of PR - automatically enable c++14 and root7 on most platforms - only centos is excluded.
I need to add extra checks and prevent reassignment of CMAKE_CXX_STANDARD when it already specified.

File: jenkins/root-build.cmake (hunk 5 above)
Comment (linev) [reply jenkins/root-build.cmake]: Finally, c++ standard should be ok.
By default it was forced to c++11.
Now with root7 it will be forced to c++14 (beside centos)
And as before, you could provide -DCMAKE_CXX_STANDARD=17 in your custom flags.
These flags placed at the end of cmake options list and will overrule default settings

File: jenkins/root-build.cmake (hunk 2 above)
Comment (linev) [reply jenkins/root-build.cmake]: Resolved

**Comment by linev** [comment]
Yes, seems to be your changes now working
